cpp/moderncpp.Rmd

```diff
@@ -809,28 +868,38 @@ int main(){
+This is an array where we pass the size of the array at run-time.
+We can also resize the vector by adding elements. [Reference Page](https://en.cppreference.com/w/cpp/container/vector).
 
-It has both a size and a capacity, size is the number of current elements and capacity is the current size of the underlying c array. When the size reaches capacity, c++ will allocate more memory and increase the capacity. Make sure that you try to reduce the number of time that the capacity is changed, one way to do this is to reserve space when you plan to add more than one element.
+It has both a size and a capacity.
+Size is the number of current elements and capacity is the current size of the underlying c array.
+When the size reaches capacity, c++ will allocate more memory and increase the capacity.
+Make sure that you try to reduce the number of times that the capacity is changed, to prevent unnecessary re-allocation performance overhead.
+One way to do this is to reserve space by specifying a size in the constructor.
 
-Removing and adding elements near the start of the vector is slow, see [Reference Page](https://en.cppreference.com/w/cpp/container/vector) for time complexities, use a deque when you plan to do this often.
+Removing and adding elements near the start of the vector is slow (see [Reference Page](https://en.cppreference.com/w/cpp/container/vector) for time complexities).
+Use a deque when you plan to do this often.
 
-If you have done java, this is similar to `ArrayList`.
+If you know Java, this is similar to `ArrayList`.
+It's C++'s basic dynamically sized array and it's the first data structure you should think of in most situations.
 
 ## Deque
 
-A `std::deque` implements a deque. The expansion of capacity and insertion or removal of elements at the ends is much faster than a vector, but takes up more memory and access is slower. Usually a vector is better as we usually do more reads than insertion and removals.
+A `std::deque` implements a deque (double-ended queue).
+The expansion of capacity and insertion or removal of elements at the ends is much faster than a vector, but takes up more memory, and accesses are slower.
+Usually, a vector is better as we often do more reads than insertion and removals.
 
 ## Map
 
-This is an associative data structure, it associates values to keys. It can either be a red-black tree, [std::map](https://en.cppreference.com/w/cpp/container/map), or a hash map, [std::unordered_map](https://en.cppreference.com/w/cpp/container/unordered_map).
+This is an associative data structure which associates values to keys.
+It can either be a red-black tree ([std::map](https://en.cppreference.com/w/cpp/container/map)), or a hash map ([std::unordered_map](https://en.cppreference.com/w/cpp/container/unordered_map)).
+Maps are bad for cache-locality and they should be avoided where possible in performance critical code.
```

Comment:
Just say they are slow, no one will know about cache locality.

Reply:
I think that's implied by the rest of the sentence. I think this detail is worth keeping.